app/src/store/playground/playgroundStore.tsx

```diff
@@ -192,6 +207,7 @@ export const createPlaygroundStore = (
           return instance;
         }),
       });
+      get().calculateVariables();
```

Comment:
This may be heavy handed, but with the size of our data I don't see any performance problems with this.

There may be a better place to hook this logic into however that I'm not familiar with.

Reply:
ahh this is basically calculating variables everytime the instance changes?

I was just thinking over whether it's worth it to have some abstractions like updateMessage and stuff as opposed to everything going through updateInstance. Doesn't seem necessary now, but if we did this could go there

Reply:
Yeah I was going back and forth on that, I wasn't sure what all of the operations were that flow through updateInstance.

Rather than writing bespoke abstractions for each update operation and making sure we calculate variables in all of the correct abstractions, I figured this would catch everything for not much perf cost.

Otherwise yeah I agree with you. Additionally we could store variables per-instance and merge them all together before display so that we only have to calculate variables in a single instance at a time.